reflex/config.py

```diff
@@ -126,130 +124,135 @@ def get_url(self) -> str:
         """Initialize the config values.
 
-        If db_url is not provided gets it from db_config.
-
         Args:
             *args: The args to pass to the Pydantic init method.
             **kwargs: The kwargs to pass to the Pydantic init method.
         """
-        if "db_url" not in kwargs and "db_config" in kwargs:
-            kwargs["db_url"] = kwargs["db_config"].get_url()
-
         super().__init__(*args, **kwargs)
 
-        # set overriden class attribute values as os env variables to avoid losing them
-        for key, value in dict(self).items():
-            key = key.upper()
-            if (
-                key.startswith("_")
-                or key in os.environ
-                or (value is None and key != "DB_URL")
-            ):
-                continue
-            os.environ[key] = str(value)
-
-        # Avoid overriding if env_path is not provided or does not exist
-        if self.env_path is not None and os.path.isfile(self.env_path):
-            load_dotenv(self.env_path, override=self.override_os_envs)  # type: ignore
-            # Recompute constants after loading env variables
-            importlib.reload(constants)
-            # Recompute instance attributes
-            self.recompute_field_values()
-
-    def recompute_field_values(self):
-        """Recompute instance field values to reflect new values after reloading
-        constant values.
+        # Check for deprecated values.
+        self.check_deprecated_values(**kwargs)
+
+        # Update the config from environment variables.
+        self.update_from_env()
+
+    @staticmethod
+    def check_deprecated_values(**kwargs):
+        """Check for deprecated config values.
+
+        Args:
+            **kwargs: The kwargs passed to the config.
+
+        Raises:
+            ValueError: If a deprecated config value is found.
+        """
+        if "db_config" in kwargs:
+            raise ValueError("db_config is deprecated - use db_url instead")
+        if "admin_dash" in kwargs:
+            raise ValueError(
+                "admin_dash is deprecated in the config - pass it as a param to rx.App instead"
+            )
+        if "env_path" in kwargs:
+            raise ValueError(
+                "env_path is deprecated - use environment variables instead"
+            )
+
+    def update_from_env(self):
+        """Update the config from environment variables.
+
+
+        Raises:
+            ValueError: If an environment variable is set to an invalid type.
         """
-        for field in self.get_fields():
-            try:
-                if field.startswith("_"):
-                    continue
-                setattr(self, field, getattr(constants, f"{field.upper()}"))
-            except AttributeError:
-                pass
+        # Iterate over the fields.
+        for key, field in self.__fields__.items():
+            # The env var name is the key in uppercase.
+            env_var = os.environ.get(key.upper())
```

Comment:
oh, we are doing this? is this temporary or will it be supported going forward for current and future config vars?